client/state/plugins/upload/actions.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Internal dependencies
+ */
+import {
+	PLUGIN_UPLOAD,
+	PLUGIN_UPLOAD_UPDATE_PROGRESS,
+	PLUGIN_UPLOAD_COMPLETE,
+	PLUGIN_UPLOAD_ERROR,
+} from 'state/action-types';
+
+/**
+ * Upload a plugin to a site.
+ *
+ * @param {number} siteId site ID
+ * @param {File} file the plugin zip to upload
+ * @return {Object} action object
+ */
+export function uploadPlugin( siteId, file ) {
```

Comment:
I'm not convinced on the value of adding tests just for the action types, but tests that use them all are coming in #15832.